### PR TITLE
build compat layer software.eessi.io/2023.06

### PR DESCRIPTION
**Pull request by trz42**
- changed repository name to software.eessi.io
- fixed creation of tarball at the end of `bot/build.sh`
  - should be improved to handle errors when some tmp directory couldn't be determined
- added `bot/check-build.sh` that prepares `.result` file including a status update for the PR comment
- added a bit more debug output in `install_compatibility_layer.sh`, should help with setting up debug sessions

Did not change CI `pilot_repo.yml` yet.

### CODE REVIEW

**Comment by trz42** [comment]
bot: build arch:generic repo:compat

**Comment by trz42** [comment]
The building "worked", the task that failed was the ReFrame test at the end. This tests (among other things) if a `startprefix` under the correct prefix is found. It wasn't. It seems, re-using the repo config (repo name) from `eessi-2023.06-compat` didn't work. The old prefix was used.

Reconfigured the bot to be able to build for two repositories `eessi.org-2023.06-compat` (renamed from `eessi-2023.06-compat`) and `eessi.io-2023.06-compat` (for the new one).

bot: show_config

**Comment by trz42** [comment]
bot: build arch:generic repo:eessi.io-2023.06-compat

**Comment by boegel** [comment]
@trz42 @bedroge This can be merged too now right, since the compat layer has already been ingested (and even package updates have been installed, see #193)?

File: bot/check-build.sh [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+[[ ${VERBOSE} -ne 0 ]] && echo ">> cfg[architecture][software_subdir] = ${cpu_target_arch}"
+
+host_arch=$(uname -m)
+eessi_arch=${cpu_target_arch:-${host_arch}}
+# eessi_os=linux
+job_version=$(cfg_get_value "repository" "repo_version")
+eessi_version=${job_version:-2023.09}
+# job_repo=$(cfg_get_value "repository" "repo_name")
+# eessi_repo=${job_repo:-pilot.nessi.no}
+# tar_topdir=/cvmfs/${eessi_repo}/versions
```
Comment (bedroge) [inline bot/check-build.sh:94]: Can these commented lines be removed?

File: bot/check-build.sh [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+if [[ ${failed_eq_zero} -eq 0 ]] && [[ ${found_tarballs} -eq 0 ]]; then
+    status="SUCCESS"
+    summary=":grin: SUCCESS"
+else
+    status="FAILURE"
+    summary=":cry: FAILURE"
+fi
+
+# TODO adjust format to what NESSI bot uses
```
Comment (bedroge) [inline bot/check-build.sh:226]: Not sure what needs to happen here, but this mentions NESSI? Can it be removed, or is there still something that needs to be changed?

**Review by bedroge — CHANGES_REQUESTED** [review]
@trz42  Just two small comments, but other than that I guess it's ready to be merged?

**Comment by boegel** [comment]
I prefer to have this merged, to avoid surprises going forward, since this is active in `software.eessi.io/versions/2023.06/compat`.

If any cleanup is still required, it can be done in a subsequent PR...

**Review by boegel — APPROVED** [review]

**Pull request merged**